components/ChallengeForm.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+
+import {
+  StyledForm,
+  StyledInput,
+  StyledBox,
+  StyledRange,
+  StyledButton,
+} from "../components/StyledComponents";
+
+// create a callback function that is called when the form is submitted
+// the callback function should create a new challenge object
+// and call the onAddChallenge function that is passed as a prop
+
+async function persistAddedChallenge(challenge) {
```

Comment:
I don't understand whats every Detail here, but the preview(codesandbox) throws an Runtime-Error, when trying to fetch.

I do not really understand why you are using an API, and fetchting/posting at this point. 

Couldn't you just use a State to Store your Data persistent? (LocalStorage)

Reply:
Ah thanks for the error info!
You're right, I guess that would be much simpler. Correct me if I'm wrong but I thought I need to use an api to use the CRUD operations more easily to change & delete the data in the next step